libtenzir/src/curl.cpp

```diff
@@ -226,6 +226,17 @@ auto multi::run(std::chrono::milliseconds timeout) -> caf::expected<size_t> {
   if (result != code::ok) {
     return to_error(result);
   }
+  auto queued = int{};
+  auto* msg = curl_multi_info_read(multi_.get(), &queued);
+  if (msg) {
+    auto response_code = int64_t{0};
+    curl_easy_getinfo(msg->easy_handle, CURLINFO_RESPONSE_CODE, &response_code);
```

Comment:
I think you need to check the return code and perform the subsequent response code check only after you got `code::ok`.